jf_agent/main.py

```diff
@@ -439,23 +441,48 @@ def print_all_jira_fields(config, jira_connection):
 @diagnostics.capture_timing()
 @agent_logging.log_entry_exit(logger)
 def download_data(config, endpoint_git_instance_info, jira_connection, git_connection):
+    write_file(config.outdir, 'status', config.compress_output_files, [])
     if jira_connection:
         load_and_dump_jira(config, jira_connection)
 
     if git_connection:
         try:
             if config.git_provider == 'bitbucket_server':
                 load_and_dump_bb(config, endpoint_git_instance_info, git_connection)
+                with open(f'{config.outdir}/status.json', 'r') as status:
+                    status_json = json.load(status)
+                    body = {
+                        'type': 'Git',
+                        'status': 'success'
+                    }
+                    status_json.append(body)
+                    write_file(config.outdir, 'status', config.compress_output_files, status_json)
             elif config.git_provider == 'github':
                 load_and_dump_github(config, endpoint_git_instance_info, git_connection)
+                with open(f'{config.outdir}/status.json', 'r') as status:
+                    status_json = json.load(status)
+                    body = {
+                        'type': 'Git',
+                        'status': 'success'
+                    }
+                    status_json.append(body)
+                    write_file(config.outdir, 'status', config.compress_output_files, status_json)
         except Exception as e:
             agent_logging.log_and_print(
                 logger,
                 logging.ERROR,
                 f'Failed to download {config.git_provider} data:\n{e}',
-                exc_info=True,
+                exc_info=True
             )
 
+            with open(f'{config.outdir}/status.json', 'r') as status:
+                status_json = json.load(status)
+                body = {
+                    'type': 'Git',
+                    'status': 'failed'
+                }
+                status_json.append(body)
+                write_file(config.outdir, 'status', config.compress_output_files, status_json)
```

Comment:
the 3 blocks here are almost identical -- maybe create a helper function that takes in parameters, or else assign local vars and then execute the code once at the end of the function